drop-transfer/src/ws/server/v4.rs

```diff
@@ -651,6 +656,20 @@ impl handler::Downloader for Downloader {
         }))
         .await
     }
+
+    async fn validate(&mut self, path: &Hidden<PathBuf>) -> crate::Result<()> {
+        let csum = tokio::task::block_in_place(|| {
+            let file = std::fs::File::open(&path.0)?;
+            let csum = file::checksum(&mut io::BufReader::new(file))?;
+            crate::Result::Ok(csum)
+        })?;
+
+        if self.full_csum.get().await == csum {
+            Ok(())
+        } else {
+            Err(crate::Error::CheckSumMismatch)
+        }
```

Comment:
Mega minor: To fall in-line with the refactor in `ws/server/mod.rs`, but can ignore this one

```suggestion
        if self.full_csum.get().await != csum {
            return Err(crate::Error::CheckSumMismatch)
        }
        
        Ok(())
```

Reply:
Applied